Build failure on systems with large GID/UID

Hello @lvca,

cc

When compiling ArcadeDB (with `mvn clean install -DskipTests -X`), got this error every time : 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.8.0:single (full) on project arcadedb-package: 
Execution full of goal org.apache.maven.plugins:maven-assembly-plugin:3.8.0:single failed: 
group id '..............' is too big ( > 2097151 ). Use STAR or POSIX extensions to overcome this limit
```

I resolved this by switching : 

1) In the **`package/pom.xml`**: replaced all occurrences of `gnu` by `posix`

2) In the **`pom.xml`**, the`<pluginManagement>`, updating : 

```
                <plugin>
                    <artifactId>maven-assembly-plugin</artifactId>
                    <version>${maven-assembly-plugin.version}</version>
                        <configuration>
                            <tarLongFileMode>posix</tarLongFileMode>
                        </configuration>
                </plugin>
```

Could you integrate a / this fix ?

Have a nice day Luca!

V

--- a/README.md
+++ b/README.md
@@ -200,6 +200,37 @@ Docker images are available on ghcr.io too:
 docker pull ghcr.io/arcadedata/arcadedb:26.3.1-java17
 ```
 
+### Building and Testing
+
+Build the entire project (skipping tests):
+
+```bash
+mvn clean install -DskipTests
+```
+
+Run the full test suite:
+
+```bash
+mvn test
+```
+
+Some tests are tagged to indicate their cost:
+
+- `slow` - functional tests that take noticeably long (large batches, multi-second elapsed time, big payloads)
+- `benchmark` - microbenchmarks not intended for regular CI runs
+
+To skip these and run only the fast tests:
+
+```bash
+mvn test -DexcludedGroups="slow,benchmark"
+```
+
+To run only a specific tag (e.g. benchmark tests in isolation):
+
+```bash
+mvn test -Dgroups="benchmark"
+```
+
 ### Community
 
 Join our growing community around the world, for ideas, discussions and help regarding ArcadeDB.

--- a/package/pom.xml
+++ b/package/pom.xml
@@ -59,7 +59,7 @@
                                 <descriptor>./src/main/assembly/full.xml</descriptor>
                             </descriptors>
                             <ignoreDirFormatExtensions>false</ignoreDirFormatExtensions>
-                            <tarLongFileMode>gnu</tarLongFileMode>
+                            <tarLongFileMode>posix</tarLongFileMode>
                         </configuration>
                     </execution>
                     <execution>
@@ -75,7 +75,7 @@
                                 <descriptor>./src/main/assembly/minimal.xml</descriptor>
                             </descriptors>
                             <ignoreDirFormatExtensions>false</ignoreDirFormatExtensions>
-                            <tarLongFileMode>gnu</tarLongFileMode>
+                            <tarLongFileMode>posix</tarLongFileMode>
                         </configuration>
                     </execution>
                     <execution>
@@ -91,7 +91,7 @@
                                 <descriptor>./src/main/assembly/headless.xml</descriptor>
                             </descriptors>
                             <ignoreDirFormatExtensions>false</ignoreDirFormatExtensions>
-                            <tarLongFileMode>gnu</tarLongFileMode>
+                            <tarLongFileMode>posix</tarLongFileMode>
                         </configuration>
                     </execution>
                     <execution>
@@ -107,7 +107,7 @@
                                 <descriptor>./src/main/assembly/base.xml</descriptor>
                             </descriptors>
                             <ignoreDirFormatExtensions>false</ignoreDirFormatExtensions>
-                            <tarLongFileMode>gnu</tarLongFileMode>
+                            <tarLongFileMode>posix</tarLongFileMode>
                         </configuration>
                     </execution>
                 </executions>

--- a/package/pom.xml
+++ b/package/pom.xml
@@ -59,7 +59,6 @@
                                 <descriptor>./src/main/assembly/full.xml</descriptor>
                             </descriptors>
                             <ignoreDirFormatExtensions>false</ignoreDirFormatExtensions>
-                            <tarLongFileMode>posix</tarLongFileMode>
                         </configuration>
                     </execution>
                     <execution>
@@ -75,7 +74,6 @@
                                 <descriptor>./src/main/assembly/minimal.xml</descriptor>
                             </descriptors>
                             <ignoreDirFormatExtensions>false</ignoreDirFormatExtensions>
-                            <tarLongFileMode>posix</tarLongFileMode>
                         </configuration>
                     </execution>
                     <execution>
@@ -91,7 +89,6 @@
                                 <descriptor>./src/main/assembly/headless.xml</descriptor>
                             </descriptors>
                             <ignoreDirFormatExtensions>false</ignoreDirFormatExtensions>
-                            <tarLongFileMode>posix</tarLongFileMode>
                         </configuration>
                     </execution>
                     <execution>
@@ -107,7 +104,6 @@
                                 <descriptor>./src/main/assembly/base.xml</descriptor>
                             </descriptors>
                             <ignoreDirFormatExtensions>false</ignoreDirFormatExtensions>
-                            <tarLongFileMode>posix</tarLongFileMode>
                         </configuration>
                     </execution>
                 </executions>

--- a/pom.xml
+++ b/pom.xml
@@ -201,7 +201,7 @@
                 <version>${maven-surefire-plugin.version}</version>
                 <configuration>
                     <!-- ADD THIS LINE TO SKIP BENCHMARKS BY DEFAULT -->
-                    <excludedGroups>benchmark</excludedGroups>
+<!--                    <excludedGroups>benchmark</excludedGroups>-->
                     <properties>
                         <!-- Work around. Surefire does not include enough
                              information to disambiguate between different

--- a/pom.xml
+++ b/pom.xml
@@ -210,7 +210,7 @@
                             cucumber.junit-platform.naming-strategy=long
                         </configurationParameters>
                     </properties>
-<!--                            cucumber.filter.name="not .*IT.*"-->
+                    <!--                            cucumber.filter.name="not .*IT.*"-->
                     <forkedProcessExitTimeoutInSeconds>60</forkedProcessExitTimeoutInSeconds>
                     <forkCount>1</forkCount>
                     <systemPropertyVariables>
@@ -335,7 +335,9 @@
                 <configuration>
                     <licenseMerges>
                         <!-- Group similar licenses to avoid duplicates -->
-                        <licenseMerge>Apache License 2.0|Apache 2.0|Apache License, Version 2.0|The Apache Software License, Version 2.0|Apache-2.0</licenseMerge>
+                        <licenseMerge>Apache License 2.0|Apache 2.0|Apache License, Version 2.0|The Apache Software License, Version
+                            2.0|Apache-2.0
+                        </licenseMerge>
                         <licenseMerge>MIT License|MIT|The MIT License</licenseMerge>
                         <licenseMerge>BSD 3-Clause|BSD-3-Clause|3-Clause BSD License|BSD 3-Clause License</licenseMerge>
                         <licenseMerge>BSD 2-Clause|BSD-2-Clause|2-Clause BSD License</licenseMerge>
@@ -361,6 +363,9 @@
                 <plugin>
                     <artifactId>maven-assembly-plugin</artifactId>
                     <version>${maven-assembly-plugin.version}</version>
+                    <configuration>
+                        <tarLongFileMode>posix</tarLongFileMode>
+                    </configuration>
                 </plugin>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
@@ -428,45 +433,45 @@
                     </configuration>
                 </plugin>
 
-	            <!-- gRPC Protobuf Plugin -->
-	            <plugin>
-	                <groupId>io.github.ascopes</groupId>
-	                <artifactId>protobuf-maven-plugin</artifactId>
-	                <version>${protobuf-maven-plugin.version}</version>
-	                <configuration>
-	                    <protoc>${protoc.version}</protoc>
-	                    <embedSourcesInClassOutputs>true</embedSourcesInClassOutputs>
+                <!-- gRPC Protobuf Plugin -->
+                <plugin>
+                    <groupId>io.github.ascopes</groupId>
+                    <artifactId>protobuf-maven-plugin</artifactId>
+                    <version>${protobuf-maven-plugin.version}</version>
+                    <configuration>
+                        <protoc>${protoc.version}</protoc>
+                        <embedSourcesInClassOutputs>true</embedSourcesInClassOutputs>
                         <plugins>
                             <plugin kind="binary-maven">
-	                            <groupId>io.grpc</groupId>
-	                            <artifactId>protoc-gen-grpc-java</artifactId>
-	                            <version>${grpc.version}</version>
+                                <groupId>io.grpc</groupId>
+                                <artifactId>protoc-gen-grpc-java</artifactId>
+                                <version>${grpc.version}</version>
                             </plugin>
                         </plugins>
-	                    <sourceDirectories>
-	                        <sourceDirectory>${project.basedir}/src/main/proto</sourceDirectory>
-	                        <sourceDirectory>${project.basedir}/src/generated/proto</sourceDirectory>
-	                    </sourceDirectories>
-	                    <outputDirectory>src/generated/protobuf/java</outputDirectory>
-	                    <includes>
-	                        <include>*.proto</include>
-	                        <include>**/*.proto</include>
-	                    </includes>
-	                </configuration>
-	                <executions>
-	                    <execution>
-	                        <goals>
-	                            <goal>generate</goal>
-	                        </goals>
-	                        <configuration>
-	                            <sourceDirectories>
-	                                <sourceDirectory>${project.basedir}/src/generated/proto</sourceDirectory>
-	                                <sourceDirectory>${project.basedir}/src/main/proto</sourceDirectory>
-	                            </sourceDirectories>
-	                        </configuration>
-	                    </execution>
-	                </executions>
-	            </plugin>
+                        <sourceDirectories>
+                            <sourceDirectory>${project.basedir}/src/main/proto</sourceDirectory>
+                            <sourceDirectory>${project.basedir}/src/generated/proto</sourceDirectory>
+                        </sourceDirectories>
+                        <outputDirectory>src/generated/protobuf/java</outputDirectory>
+                        <includes>
+                            <include>*.proto</include>
+                            <include>**/*.proto</include>
+                        </includes>
+                    </configuration>
+                    <executions>
+                        <execution>
+                            <goals>
+                                <goal>generate</goal>
+                            </goals>
+                            <configuration>
+                                <sourceDirectories>
+                                    <sourceDirectory>${project.basedir}/src/generated/proto</sourceDirectory>
+                                    <sourceDirectory>${project.basedir}/src/main/proto</sourceDirectory>
+                                </sourceDirectories>
+                            </configuration>
+                        </execution>
+                    </executions>
+                </plugin>
 
             </plugins>
         </pluginManagement>

